packages/core/core/test/AssetGraph.test.js

```diff
@@ -27,41 +27,40 @@ const stats = {size: 0, time: 0};
 describe('AssetGraph', () => {
   it('initialization should create one root node with edges to dependency nodes for each entry', () => {
     let graph = new AssetGraph();
-    graph.initializeGraph({
+    graph.initialize({
       targets: TARGETS,
-      entries: ['./index1', './index2'],
-      rootDir: '/'
+      entries: ['/path/to/index1', '/path/to/index2']
     });
 
-    assert(graph.nodes.has('/'));
+    assert(graph.nodes.has('@@root'));
     assert(
       graph.nodes.has(
         new Dependency({
-          moduleSpecifier: './index1',
+          moduleSpecifier: '/path/to/index1',
```

Comment:
why did these change to absolute?

Reply:
I noticed that in the initialization of the `AssetGraph` we are not even instantiating the entry deps with `sourcePath`, seemingly because the entries are all resolved to absolute paths before hand so we don't need the `sourcePath` to know where to resolve from. I guess resolving entries to absolute paths is happening in the CLI though. Should we require users of the programmatic API to pass entries as absolute paths?

Reply:
> Should we require users of the programmatic API to pass entries as absolute paths?

I think we should follow what we're doing with `cacheDir` and target dist information passed through the public api: just do what Node does and resolve things relative to pwd.

Of course, just as you'd do with `fs.readFile`, etc., you're far better off being certain and passing an absolute path.

Reply:
FWIW, while exploring options overriding node module resolution in #3071, I found that when you run `node` with an entrypoint, it invokes the resolver with a `null` sourcePath equivalent and an absolute path for the `moduleSpecifier` equivalent.